src/dsp_tools/utils/connection_live.py

```diff
@@ -162,30 +236,34 @@ def post(
         # otherwise the client can get a timeout error while the API is still processing the request
         # in that case, the client's retry will have undesired side effects (e.g. duplicated resources),
         # and the response of the original API call will be lost
-        timeout = 60
+        timeout = timeout or 60
         if not route.startswith("/"):
             route = f"/{route}"
         url = self.server + route
         headers = {}
         if jsondata:
-            headers["Content-Type"] = f"{content_type}; charset=UTF-8"
+            headers["Content-Type"] = "application/json; charset=UTF-8"
         if self.token:
             headers["Authorization"] = f"Bearer {self.token}"
 
-        response = requests.post(
-            url=url,
-            headers=headers,
+        request = partial(requests.post, url=url, headers=headers, timeout=timeout)
+        if jsondata:
             # if data is not encoded as bytes, issues can occur with non-ASCII characters,
             # where the content-length of the request will turn out to be different from the actual length
-            data=jsondata.encode("utf-8") if jsondata else None,
-            timeout=timeout,
-        )
+            data = jsondata.encode("utf-8") if jsondata else None
+            request = partial(request, data=data)
+        elif files:
+            request = partial(request, files=files)
+            logger.warning(f"POST with file: {request}")
```

Comment:
Is a file upload so dangerous that we have to warn? ;-)

```suggestion
            logger.info(f"POST with file: {request}")
```

Reply:
actually, this line should be removed entirely, just used it for diagnosing some failing tests